Client/Frontend/Browser/BrowserToolbar.swift

```diff
@@ -7,12 +7,11 @@ import UIKit
 import Snappy
 
 protocol BrowserToolbarDelegate {
-    func didClickBack()
-    func didClickForward()
-    func didLongPressBack()
-    func didLongPressForward()
-
-    func didClickBookmark()
+    func browserToolbarDidPressBack()
```

Comment:
If you *really* want to follow the UIKit conventions then the first argument should also be a reference to the object calling the delegate. I know that does not always make sense in cases where there is just one instance alive, but .. that is what UIKit does.

(It does help with state management. Specially when presenting viewcontrollers, it means that you don't have to keep a reference to the viewcontroller around. You can just use the reference that you get in your delegates.)